src/app/backing/components/AllocationBar/AllocationBarSegment.tsx

```diff
@@ -0,0 +1,72 @@
+import { useSortable } from '@dnd-kit/sortable'
+import { CSS } from '@dnd-kit/utilities'
+import { AllocationBarDragHandle } from './AllocationBarDragHandle'
+import { AllocationBarResizeHandle } from './AllocationBarResizeHandle'
+import { AllocationItem } from './types'
+import { checkerboardStyle, MIN_SEGMENT_PERCENT } from './utils'
+import { formatPercentage } from '@/lib/utils'
+
+const AllocationBarSegmentPercent = ({ value }: { value: number }) => {
+  return (
+    <span className="absolute -top-7 left-1/2 -translate-x-1/2 pointer-events-none whitespace-nowrap font-normal leading-5 text-v3-bg-accent-0 font-rootstock-sans">
+      {formatPercentage(value)}%
+    </span>
+  )
+}
+
+interface AllocationBarSegmentProps {
+  value: number
+  item: AllocationItem
+  index: number
+  isLast: boolean
+  showPercent: boolean
+  onHandleMouseDown: (idx: number) => (e: React.MouseEvent) => void
+  dragIndex: number | null
+  isDraggable: boolean
+  isResizable: boolean
+}
```

Comment:
move to types.ts?

Reply:
I wouldn't as this type isn't meant to be used externally.